operator/v1/types_dns.go

```diff
@@ -376,6 +411,20 @@ const (
 	NetworkResolverType   UpstreamType = "Network"
 )
 
+// ProtocolStrategy is a preference for the protocol to use for DNS queries.
+// +kubebuilder:validation:Enum:=ForceTCP;""
+type ProtocolStrategy string
+
+var (
+	// ProtocolStrategyDefault specifies no opinion for DNS protocol.
+	// Currently, this means that CoreDNS uses the protocol of the originating client request as the upstream protocol.
+	// Note that this behavior is subject to change.
+	ProtocolStrategyDefault ProtocolStrategy = ""
+
+	// ProtocolStrategyForceTCP instructs CoreDNS to always use TCP, regardless of the originating client's request protocol.
+	ProtocolStrategyForceTCP ProtocolStrategy = "ForceTCP"
```

Comment:
nit - I would make the literal value "TCP", as the customer doesn't need the implementation details.

Reply:
Done.